services/navigation/path.go

```diff
@@ -0,0 +1,103 @@
+// Package navigation is the service that allows you to navigate along waypoints.
+package navigation
+
+import (
+	"errors"
+
+	geo "github.com/kellydunn/golang-geo"
+	commonpb "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/service/navigation/v1"
+)
+
+var (
+	errNilSlice = errors.New("cannot convert nil slice")
+	errNilPath  = errors.New("cannot convert nil path")
+)
+
+// Path describes a series of geo points the robot will travel through.
+type Path struct {
+	destinationWaypointID string
+	geoPoints             []*geo.Point
+}
+
+// NewPath constructs a Path from a slice of geo.Points and ID.
+func NewPath(id string, geoPoints []*geo.Point) (*Path, error) {
+	if geoPoints == nil {
+		return nil, errors.New("cannot instantiate path with no geoPoints")
+	}
+	return &Path{
+		destinationWaypointID: id,
+		geoPoints:             geoPoints,
+	}, nil
+}
+
+// DestinationWaypointID returns the ID of the Path.
+func (p *Path) DestinationWaypointID() string {
+	return p.destinationWaypointID
+}
+
+// GeoPoints returns the slice of geo.Points the Path is comprised of.
+func (p *Path) GeoPoints() []*geo.Point {
+	return p.geoPoints
+}
```

Comment:
[q]: should I add methods on the `Path` struct like `SetGeoPoints` or `SetDestinationWaypointID`?

Reply:
I don't think so